lib/dependencies/CommonJsImportsParserPlugin.js

```diff
@@ -375,14 +444,192 @@ class CommonJsImportsParserPlugin {
 
 		parser.hooks.call
 			.for("require.resolve")
-			.tap("RequireResolveDependencyParserPlugin", expr => {
+			.tap("CommonJsImportsParserPlugin", expr => {
 				return processResolve(expr, false);
 			});
 		parser.hooks.call
 			.for("require.resolveWeak")
-			.tap("RequireResolveDependencyParserPlugin", expr => {
+			.tap("CommonJsImportsParserPlugin", expr => {
 				return processResolve(expr, true);
 			});
+		//#endregion
+
+		//#region Create require
+		/**
+		 * @param {CallExpressionNode} expr call expression
+		 * @returns {string|boolean} context
+		 */
+		const parseCreateRequireArguments = expr => {
+			const args = expr.arguments;
+			if (args.length !== 1) {
+				const err = new WebpackError(
+					"module.createRequire supports only one argument."
+				);
+				err.loc = expr.loc;
+				parser.state.module.addWarning(err);
+				return;
+			}
+			const createParsingError = node => {
+				const err = new WebpackError(
+					"module.createRequire() failed parsing argument."
+				);
+				err.loc = node.loc;
+				parser.state.module.addWarning(err);
+			};
+			const arg = args[0];
+			if (
+				arg.type === "MemberExpression" &&
+				arg.object.type === "MetaProperty"
+			) {
+				if (
+					arg.object.meta.type === "Identifier" &&
+					arg.object.meta.name === "import" &&
+					arg.object.property.type === "Identifier" &&
+					arg.object.property.name === "meta" &&
+					arg.property.type === "Identifier" &&
+					arg.property.name === "url"
+				) {
+					// same module context
+					return false;
+				} else {
+					createParsingError(arg);
+				}
+			} else {
+				const evaluated = parser.evaluateExpression(arg);
+				if (!evaluated.isString()) {
+					createParsingError(arg);
+					return;
+				}
+				return evaluated.string;
+			}
+		};
+
+		parser.hooks.import.tap(
```

Comment:
should we add parser option to enable such behaviour? (and enable it by default in target=node)

Reply:
Yes, we don't want to enable that by default in web target since it's a "node.js polyfill" which we have chosen to not provide by default.